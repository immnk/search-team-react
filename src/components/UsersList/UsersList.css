.root {
  margin-top: -8rem;
}

.container {
  width: 900px;
  margin: 0 auto;
  padding: 2rem;

  @media screen and (max-width: 1024px) {
    width: 720px;
  }

  @media screen and (max-width: 540px) {
    width: 100%;
  }
}

.users {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(260px, 1fr));
  grid-gap: 1.5rem;
}
