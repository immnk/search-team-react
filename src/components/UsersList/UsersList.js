import React, { useEffect } from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import { connect } from 'redux-zero/react';

import User from 'components/User';
import actions from 'store/search/actions';
import NoUserFoundImage from './no_results.png';

import css from './UsersList.css';

const UsersList = ({
  className,
  search,
  category,
  loading,
  hasData,
  data,
  fetchUsersFromBackend
}) => {
  useEffect(() => {
    fetchUsersFromBackend();
  }, []);

  function renderUsers() {
    const filteredUsers = data.filter((user) => {
      // if (search === '' && category === '') {
      //   return true;
      // }

      const regExp = new RegExp(search, 'i');
      return regExp.test(user[category]);
    });

    // user.name.toLowerCase().indexOf(search.toLowerCase()) > -1

    return filteredUsers.length > 0 ? (
      <section className={css.users}>
        {filteredUsers.map((user) => (
          <User key={user.id} user={user} />
        ))}
      </section>
    ) : (
      <figure className={css.noUser}>
        <img src={NoUserFoundImage} alt="User not found" />
        <figcaption>No users found containing {search}</figcaption>
      </figure>
    );
  }

  function renderData() {
    return data.length > 0 ? renderUsers() : <p>No users present</p>;
  }

  return (
    <section className={classNames(css.root, className)}>
      <div className={css.container}>
        {loading ? <p>Loading...</p> : false}
        {hasData ? renderData() : false}
      </div>
    </section>
  );
};

UsersList.defaultProps = {
  className: undefined,
  category: '',
  search: '',
  loading: false,
  hasData: false,
  data: []
};

UsersList.propTypes = {
  className: PropTypes.string,
  search: PropTypes.string,
  category: PropTypes.string,
  loading: PropTypes.bool,
  hasData: PropTypes.bool,
  data: PropTypes.arrayOf(PropTypes.shape({})),
  fetchUsersFromBackend: PropTypes.func.isRequired
};

const mapStateToProps = ({
  search: { search, category, loading, hasData, error, errorMessage, data }
}) => ({
  search,
  category,
  loading,
  hasData,
  error,
  errorMessage,
  data
});

export default connect(mapStateToProps, actions)(UsersList);
