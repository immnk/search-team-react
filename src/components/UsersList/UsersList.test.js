import React from 'react';
import { shallow } from 'enzyme';

import UsersList from './UsersList';

const renderUsersList = (props) => {
  const defaultProps = {
    className: 'my-classname',
    fetchUsersFromBackend: () => {},
    loading: false,
    hasData: true,
    data: [
      {
        id: 1,
        name: 'Leanne Graham',
        username: 'Bret',
        email: 'Sincere@april.biz',
        address: {
          street: 'Kulas Light',
          suite: 'Apt. 556',
          city: 'Gwenborough',
          zipcode: '92998-3874',
          geo: {
            lat: '-37.3159',
            lng: '81.1496'
          }
        },
        phone: '1-770-736-8031 x56442'
      }
    ]
  };

  return <UsersList {...defaultProps} {...props} />;
};

const shallowRenderUsersList = (props) => shallow(renderUsersList(props));

describe('Test UsersList', () => {
  it('render app', () => {
    const actual = shallowRenderUsersList();
    expect(actual).toMatchSnapshot();
  });
});
