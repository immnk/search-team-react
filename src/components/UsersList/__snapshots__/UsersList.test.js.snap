// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test UsersList render app 1`] = `
<Connect
  actions={[Function]}
  className="my-classname"
  data={
    Array [
      Object {
        "address": Object {
          "city": "Gwenborough",
          "geo": Object {
            "lat": "-37.3159",
            "lng": "81.1496",
          },
          "street": "Kulas Light",
          "suite": "Apt. 556",
          "zipcode": "92998-3874",
        },
        "email": "Sincere@april.biz",
        "id": 1,
        "name": "Leanne Graham",
        "phone": "1-770-736-8031 x56442",
        "username": "Bret",
      },
    ]
  }
  fetchUsersFromBackend={[Function]}
  hasData={true}
  loading={false}
  mapToProps={[Function]}
>
  <Component />
</Connect>
`;
